cardinal/system_test.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"testing"
+
+	"pkg.world.dev/world-engine/cardinal"
+	"pkg.world.dev/world-engine/cardinal/receipt"
+	"pkg.world.dev/world-engine/cardinal/search/filter"
+	"pkg.world.dev/world-engine/cardinal/testutils"
+	"pkg.world.dev/world-engine/cardinal/types"
+
+	"github.com/argus-labs/starter-game-template/cardinal/component"
+	"github.com/argus-labs/starter-game-template/cardinal/msg"
+)
+
+const (
+	attackMsgName = "game.attack-player"
+	createMsgName = "game.create-player"
+)
+
+// TestErrorWhenAttackTargetDoesNotExist ensures the attack message results in an error when the given target does
+// not exist. Note, message errors are stored in receipts; they are NOT returned from the relevant system.
+func TestErrorWhenAttackTargetDoesNotExist(t *testing.T) {
```

Comment:
Convention that we should follow for tests:

`Test<category>_<relevant function/type name>_<behavior that is tested>`

For example, this should be

`TestSystem_AttackSystem_ErrorWhenTargetDoesNotExist`

Reply:
done